### DISCUSSION TITLE
Data Extraction without League Id?

### DISCUSSION THREAD

**Discussion by troylee24** [cwendt94/espn-api#277] [ESPN API]
This package already extracts enough data to, for example, get the player stats for every player in the current NBA season. For someone who would want to just get this data, irrespective of a specific league, would there be any way to initialize a `League` object without a `league_id`?

Obviously, one can always just create a new personal league for development purposes and use that as their ticket for making api calls (and honestly this is a pretty clean solution), but, out of curiosity, I was just wondering if these endpoints were accessible without having to create a league.

Also, how exactly do I get a player's stats from a player's name or player id? I see that I can get the names/ids from `player_map` in the `League` class, but not sure how I would create a `Player` object from that.

**Comment by cwendt94** [comment]
That's a great question, currently how the package is setup there is no way to make the calls without a league id. I plan on exposing the underlying [ESPN request class](https://github.com/cwendt94/espn-api/blob/master/espn_api/requests/espn_requests.py#L34) that could be used to make calls without league id and get the raw data.

Creating a player object from that data would also require some new work as a lot of the fields are specific to fantasy. A general player object could be added that has the players stats and non fantasy data.
